sdk/core/azure-core-test/CMakeLists.txt

```diff
@@ -35,6 +35,11 @@ add_library (
   ${AZURE_CORE_TEST_SOURCE}
   )
 
+if (MSVC)
+  # - C6326: Google comparisons 
+  target_compile_options(azure-core-test-fw PUBLIC /wd6326)
```

Comment:
Can we fix the issue rather than dismissing the warning?
What's an example code snippet that triggers this warning?

Reply:
EXPECT_EQ(1, someVector.size())  ->  This gives a warning

a possible fix:

size_t expected = 1;
EXPECT_EQ(expected, someVector.size())  -> this is OK

The issue is every time we would want to compare size_t vs int , for instance.

Reply:
Oh :(
Yeah, it makes sense to dismiss it then. It kind of sucks that we get a warning like that.

@CaseyCarter can you share some more C++ context here to help understand the motivation behind flagging such coding patterns as a warning.

Reply:
[C6326](https://docs.microsoft.com/en-us/cpp/code-quality/c6326?view=msvc-160) isn't for comparison of values with differing signed-ness, it's warning about "potential comparison of a constant with another constant, which is redundant code." I'd also be curious to hear why someone thinks this is a common source of bugs that deserves a warning.

Reply:
@vhvb1989 do you remember the case where we were doing constant-to-constant redundant comparison in test code? That's probably something we can fix.

Reply:
Same thing,  g-test uses macros to generate the tests and the assertions.  `EXPECT_EQ(1, someVector.size())` will be converted before on pre-build to something like `if(1 == 5)`  (supposing the vector size is 5.  

This warning can't co-exists with the way gtest works.  And it is a warning for when we go to be pedantic in the things we want to catch (/w4) . On standard build, MSVC won't complain about it. We we wanted to go further.

Also, don't worry too much about what flags/warnings we disable/enable for testing. It's just the testing artifacts.  We don't disable this for the libraries